modules/data/event-hub/main.bicep

```diff
@@ -0,0 +1,331 @@
+status: (Optional) string, Enumerates the possible values for the status of the Event Hub. 'Active','Creating','Deleting','Disabled','ReceiveDisabled','Renaming','Restoring','SendDisabled','Unknown'. Default to 'Active'.
+captureDescriptionDestinationName: (Optional) string, Name for capture destination.
+captureDescriptionDestinationArchiveNameFormat: (Optional) string,  Blob naming convention for archive, e.g. {Namespace}/{EventHub}/{PartitionId}/{Year}/{Month}/{Day}/{Hour}/{Minute}/{Second}. Here all the parameters (Namespace,EventHub .. etc) are mandatory irrespective of order.
+captureDescriptionDestinationBlobContainer: (Optional) string, Blob container Name.
+captureDescriptionDestinationStorageAccountResourceId: (Optional) string, Resource ID of the storage account to be used to create the blobs.
+captureDescriptionEnabled: (Optional) boolean, A value that indicates whether capture description is enabled.
+captureDescriptionEncoding: (Optional) string, Enumerates the possible values for the encoding format of capture description.
+captureDescriptionIntervalInSeconds: (Optional) int, The time window allows you to set the frequency with which the capture to Azure Blobs will happen.
+captureDescriptionSizeLimitInBytes: (Optional) int, The size window defines the amount of data built up in your Event Hub before an capture operation.
+captureDescriptionSkipEmptyArchives: (Optional) boolean,  A value that indicates whether to Skip Empty Archives.
+roleAssignments: (Optional) Array, Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.
+''')
+param eventHubs object = {}
+
+@description('Optional. Authorization Rules for the Event Hub Namespace.')
+param namespaceAuthorizationRules object = {}
+
+@description('Optional. Role assignments for the namespace.')
+param namespaceRoleAssignments object = {}
+
+@description('Optional. The disaster recovery config for the namespace.')
+param disasterRecoveryConfigs object = {}
+
+@description('Optional. The Diagnostics Settings config for the namespace.')
+param diagnosticSettings object = {}
+
+@description('Optional. Authorization Rules for the Event Hub .')
+param eventHubAuthorizationRules object = {}
+
+@description('Optional. consumer groups for the Event Hub .')
+param consumerGroups object = {}
+
+@description('Define Private Endpoints that should be created for Azure Container Registry.')
+param privateEndpoints array = []
+
+@description('Toggle if Private Endpoints manual approval for Azure Container Registry should be enabled.')
+param privateEndpointsApprovalEnabled bool = false
+
```

Comment:
```suggestion
param eventHubNamespaces object = {
  'evns-${uniqueString(location)}': {
    sku: 'Standard'
    capacity: 1
    zoneRedundant: false
    isAutoInflateEnabled: false
    maximumThroughputUnits: 0
    disableLocalAuth: false
    kafkaEnabled: true
  }
}

```

Reply:
can you try this out? my IDE says its allowed...

Reply:
Added other default values